pkg/expr/sql/frame_db_conv.go

```diff
@@ -137,275 +144,155 @@ func convertDataType(fieldType data.FieldType) mysql.Type {
-
-	// ----------------------------
-	// Int64 / Nullable Int64
-	// ----------------------------
-	case data.FieldTypeInt64:
-		v, ok := val.(int64)
-		if !ok {
-			return nil, fmt.Errorf("unexpected value type for interface %v of type %T, expected int64", val, val)
+			if fieldType.Nullable() {
+				return &v, nil
+			}
+			return v, nil
 		}
-		return v, nil
+		return nil, fmt.Errorf("unexpected value type %v of type %T, expected uint32", val, val)
 
-	case data.FieldTypeNullableInt64:
-		vP, ok := val.(*int64)
-		if ok {
-			return vP, nil
-		}
+	case data.FieldTypeInt64, data.FieldTypeNullableInt64:
 		v, ok := val.(int64)
 		if ok {
-			return &v, nil
-		}
-		return nil, fmt.Errorf("unexpected value type for interface %v of type %T, expected int64 or *int64", val, val)
-
-	// ----------------------------
-	// Uint64 / Nullable Uint64
-	// ----------------------------
-	case data.FieldTypeUint64:
-		v, ok := val.(uint64)
-		if !ok {
-			return nil, fmt.Errorf("unexpected value type for interface %v of type %T, expected uint64", val, val)
+			if fieldType.Nullable() {
+				return &v, nil
+			}
+			return v, nil
 		}
-		return v, nil
+		return nil, fmt.Errorf("unexpected value type %v of type %T, expected int64", val, val)
 
-	case data.FieldTypeNullableUint64:
-		vP, ok := val.(*uint64)
-		if ok {
-			return vP, nil
-		}
+	case data.FieldTypeUint64, data.FieldTypeNullableUint64:
 		v, ok := val.(uint64)
 		if ok {
-			return &v, nil
+			if fieldType.Nullable() {
+				return &v, nil
+			}
+			return v, nil
 		}
-		return nil, fmt.Errorf("unexpected value type for interface %v of type %T, expected uint64 or *uint64", val, val)
+		return nil, fmt.Errorf("unexpected value type %v of type %T, expected uint64", val, val)
 
-	// ----------------------------
-	// Float64 / Nullable Float64
-	// ----------------------------
-	case data.FieldTypeFloat64:
-		// Accept float64 or decimal.Decimal, convert decimal.Decimal -> float64
-		if v, ok := val.(float64); ok {
-			return v, nil
+	case data.FieldTypeFloat32, data.FieldTypeNullableFloat32:
+		if fv, ok := val.(float64); ok {
+			f32 := float32(fv)
+			if fieldType.Nullable() {
+				return &f32, nil
+			}
+			return f32, nil
 		}
 		if d, ok := val.(decimal.Decimal); ok {
-			return d.InexactFloat64(), nil
+			f32 := float32(d.InexactFloat64())
+			if fieldType.Nullable() {
+				return &f32, nil
+			}
+			return f32, nil
 		}
-		return nil, fmt.Errorf("unexpected value type for interface %v of type %T, expected float64 or decimal.Decimal", val, val)
+		return nil, fmt.Errorf("unexpected value type %v of type %T, expected float64 or decimal.Decimal", val, val)
 
-	case data.FieldTypeNullableFloat64:
-		// Possibly already *float64
-		if vP, ok := val.(*float64); ok {
-			return vP, nil
-		}
-		// Possibly float64
-		if v, ok := val.(float64); ok {
-			return &v, nil
+	case data.FieldTypeFloat64, data.FieldTypeNullableFloat64:
+		if fv, ok := val.(float64); ok {
```

Comment:
Suggest renaming `fv` to `v` for consistency with other code which shares this structure?

Reply:
This one is different since it has an additional case below it for decimal.

Reply:
oh I see casing, if not `v` should be `fV`.